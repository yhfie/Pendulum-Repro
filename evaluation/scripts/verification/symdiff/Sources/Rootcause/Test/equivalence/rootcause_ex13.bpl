//scalar_func.bpl
//
// Example with equivalence checking 
// All functions are over scalars
//

function g(a:int, b:int):int;

//
procedure EQ_A(index:int, h:[int]int) {
  var h1:[int]int, h2:[int]int;
  //Version 1
  h1 := h;
  h1[index] := g(h1[index],1);


  //this is the marker that is understood by Rootcause
  goto AA_INSTR_EQ_BODY$1;
  AA_INSTR_EQ_BODY$1: 
   

  //Version 2
  h2 := h;
  h2[index] := g(h2[index+1],1);

  assert(forall i:int :: h1[i] == h2[i]);
}

