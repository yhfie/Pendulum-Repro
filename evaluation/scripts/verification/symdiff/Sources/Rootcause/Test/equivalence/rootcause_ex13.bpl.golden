BENCHMARK:rootcause_ex13.bpl(ERRORFREE)
Processing file rootcause_ex13.bpl
############# Implementation = EQ_A #################
Cause ==>
leftAssignment: h1[index] := g(h1[index], 1);
rightAssignment: h2[index] := g(h2[index + 1], 1);
